hq/app/aws/ec2/EC2.scala

```diff
@@ -50,19 +50,27 @@ object EC2 {
     }
   }
 
-  def flaggedSgsForAccount(account: AwsAccount)(implicit ec: ExecutionContext): Attempt[List[(SGOpenPortsDetail, Set[SGInUse])]] = {
+  def flaggedSgsForAccount(account: AwsAccount, refresh: Boolean = true)(implicit ec: ExecutionContext): Attempt[List[(SGOpenPortsDetail, Set[SGInUse])]] = {
     val supportClient = TrustedAdvisor.client(account)
     for {
+      _  <- if (refresh) TrustedAdvisorSGOpenPorts.refreshSGOpenPorts(supportClient) else Attempt.Right(Unit)
```

Comment:
Could the refresh be done from the call site, rather than here? This mixes the concerns of refreshing the report with fetching the flagged SGs. The fact that it takes `refresh` as a parameter suggests you think it's the call-site that should decide whether to refresh or not. In that case, is it true that having the caller refresh in the for comp of "things that need to be done" would be better than hiding it in this function?

Reply:
Another reason for separating them is that I think we will probably move the refresh commands into a service that runs separately from user-generated web requests. While doing speculative work is always bad, the fact that it also simplifies this code means that might lend some weight to exposing the refresh as a single `Attempt` call separate to fetching the SGs for an account(s).

Reply:
did changes to refresh from the controller